optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/BavetConstraintSession.java

```diff
@@ -115,7 +115,7 @@ public void update(Object fact) {
 
     @Override
     public void retract(Object fact) {
-        List<BavetFromUniTuple<Object>> tupleList = fromTupleListMap.get(fact);
+        List<BavetFromUniTuple<Object>> tupleList = fromTupleListMap.remove(fact);
```

Comment:
I think I found (and fixed) a bug in Bavet.

Reply:
Agreed. nice catch!

Can we have a unit test? :) (other PR is fine)